### PR TITLE
[Test] Replace all mock authentication with concrete objects

### PR DESCRIPTION
**Pull request by ywangd**
Changes in this PR enables marking the Authentication class to final in
a future PR as part of the overall plan for closing down Authentication.

Relates: #86206

### CODE REVIEW

**Comment by ywangd** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java [hunk 1]
```diff
@@ -403,15 +409,15 @@ public void testKibanaSystemRole() {
...
-            kibanaRole.cluster().check(CreateApiKeyAction.NAME, new GetApiKeyRequest(null, null, null, null, true), authentication),
+            kibanaRole.cluster().check(GetApiKeyAction.NAME, new GetApiKeyRequest(null, null, null, null, true), authentication),
             is(true)
         );
         final QueryApiKeyRequest queryApiKeyRequest = new QueryApiKeyRequest();
-        assertThat(kibanaRole.cluster().check(CreateApiKeyAction.NAME, queryApiKeyRequest, authentication), is(false));
+        assertThat(kibanaRole.cluster().check(QueryApiKeyAction.NAME, queryApiKeyRequest, authentication), is(false));
         queryApiKeyRequest.setFilterForCurrentUser();
-        assertThat(kibanaRole.cluster().check(CreateApiKeyAction.NAME, queryApiKeyRequest, authentication), is(true));
+        assertThat(kibanaRole.cluster().check(QueryApiKeyAction.NAME, queryApiKeyRequest, authentication), is(true));
```
Comment (albertzaharovits) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java:421]: 👍

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-1-fips

**Pull request merged**